lib/kubernetes-deploy/runner_task.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+require 'tempfile'
+
+require 'kubernetes-deploy/kubeclient_builder'
+require 'kubernetes-deploy/ui_helpers'
+require 'kubernetes-deploy/kubectl'
+
+module KubernetesDeploy
+  class RunnerTask
+    include KubernetesDeploy::KubeclientBuilder
+    include Kubectl
+
+    class FatalTaskRunError < FatalDeploymentError; end
+
+    def initialize(namespace:, context:, logger: KubernetesDeploy.logger)
+      @logger = logger
+      @namespace = namespace
+      @kubeclient = build_v1_kubeclient(context)
+      @context = context
+    end
+
+    def run(task_template:, command:, args:, env_vars: [])
+      validate_configuration(task_template, @namespace, args)
+
+      raw_template = get_template(task_template, namespace: @namespace)
+      rendered_template = build_pod_template(raw_template, command, args, env_vars)
+
+      KubernetesDeploy.logger.info("Starting task runner pod #{rendered_template.metadata.name}")
+
+      @kubeclient.create_pod(rendered_template)
+    end
+
+    private
+
+    def validate_configuration(task_template, namespace, args)
+      errors = []
+
+      if task_template.blank?
+        errors << "Task template name can't be nil"
+      end
+
+      if args.nil?
+        errors << "Args can't be nil"
+      end
+
+      if namespace.blank?
+        errors << "Namespace can't be empty"
+      end
+
+      raise FatalTaskRunError, "Configuration invalid: #{errors.join(', ')}" unless errors.empty?
+    end
+
+    def get_template(template_name, namespace:)
+      KubernetesDeploy.logger.info("Fetching task runner pod template: '#{template_name}' in namespace: '#{namespace}'")
+
+      pod_template = begin
+        @kubeclient.get_pod_template(template_name, namespace)
```

Comment:
We are actually using a `PodTemplate` so it will grab the appropriate `PodTemplate.